python/ray/data/_internal/execution/interfaces/op_runtime_metrics.py

```diff
@@ -96,18 +107,35 @@ def extra_metrics(self) -> Dict[str, Any]:
         """Return a dict of extra metrics."""
         return self._extra_metrics
 
-    def as_dict(self):
+    def as_dict(self, metrics_only: bool = False):
         """Return a dict representation of the metrics."""
         result = []
         for f in fields(self):
             if f.metadata.get("export", True):
-                if not self._is_map and f.metadata.get("map_only", False):
+                if (not self._is_map and f.metadata.get("map_only", False)) or (
+                    metrics_only and not f.metadata.get("export_metric", False)
+                ):
                     continue
                 value = getattr(self, f.name)
                 result.append((f.name, value))
+
+        resource_usage = self._op.current_resource_usage()
```

Comment:
let's add a TODO here, we should record the resource usage directly in `OpRuntimeMetrics` and avoid calling `self._op.current_resource_usage()`